bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/common/TrackingCallable.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2014,2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.internal.common;
+
+import java.lang.reflect.Method;
+import java.util.concurrent.Callable;
+
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Wraps a {@link Callable} and tracks the executing thread.
+ *
+ * @author Simon Kaufmann - initial contribution and API.
+ *
+ */
+class TrackingCallable implements Callable<Object> {
```

Comment:
Any reason not to mark this class as `@NonNullByDefault`?

Reply:
Yes, with `@NonNullByDefault` and `@Override @Nullable public Object call() {` I get the following error:

```
The return type is incompatible with '@NonNull Object' returned from Callable<Object>.call() (mismatching null constraints)```